### PR TITLE
Integrating Peter Fleury's HD44780 LCD library

### PR DESCRIPTION
**Pull request by maxaudron**
I needed HD44780 support for a project I'm working on so i integrated Peter Fleury's library.
It is under GNU GPL so i think this should be ok. 

I added a switch to rules.mk so it only compiles when needed and created a documentation page.

### CODE REVIEW

**Comment by drashna** [comment]
GNU General Public License Version 3, right?  (the linked page indicates that, but the file headers should probably state that explicitly)

Also, the c and h files for this should probably go in `/drivers/avr/` specifically, along with the other stuff.

And for the docs, you'll want to add a link to the `features.md` page too. So that it properly shows up.

File: common_features.mk [hunk 1]
```diff
@@ -199,7 +199,7 @@ endif
 
 
 ifeq ($(strip $(HD44780_ENABLE)), yes)
-    SRC += $(QUANTUM_DIR)/hd44780.c
+    SRC += drivers/avr/hd44780.c
```
Comment (drashna) [inline common_features.mk]: Not sure it matters, but the absolute path isn't necessary here.  It should be picked up just fine (like the ws2812.c and other drivers

Comment (maxaudron) [reply common_features.mk]: Yeah i wasn't sure so i added it. In my test project it compiled fine this way so i guess its staying.

**Comment by jackhumbert** [comment]
Awesome - thanks!

**Pull request merged**